test/e2e-go/restAPI/restClient_test.go

```diff
@@ -2057,7 +2067,57 @@ main_l6:
 int 1
 return`
 
-func TestMaxDepthAppWithPCTrace(t *testing.T) {
+func goValuesToStackValues(goValues ...interface{}) *[]model.StackValue {
+	if len(goValues) == 0 {
+		return nil
+	}
+
+	boolToUint64 := func(b bool) uint64 {
+		if b {
+			return 1
+		}
+		return 0
+	}
+
+	modelValues := make([]model.StackValue, len(goValues))
+	for i, goValue := range goValues {
+		switch converted := goValue.(type) {
+		case []byte:
+			modelValues[i] = model.StackValue{
+				Type:  uint64(basics.TealBytesType),
+				Bytes: &converted,
+			}
+		case bool:
+			convertedUint := boolToUint64(converted)
+			modelValues[i] = model.StackValue{
+				Type: uint64(basics.TealUintType),
+				Uint: valToNil(&convertedUint),
+			}
+		case int:
+			convertedUint := uint64(converted)
+			modelValues[i] = model.StackValue{
+				Type: uint64(basics.TealUintType),
+				Uint: valToNil(&convertedUint),
+			}
+		case basics.AppIndex:
+			convertedUint := uint64(converted)
+			modelValues[i] = model.StackValue{
+				Type: uint64(basics.TealUintType),
+				Uint: valToNil(&convertedUint),
+			}
+		case uint64:
+			modelValues[i] = model.StackValue{
+				Type: uint64(basics.TealUintType),
+				Uint: valToNil(&converted),
+			}
+		default:
+			panic("unexpected type inferred from interface{}")
+		}
+	}
+	return toPtr(modelValues)
```

Comment:
Why is `toPtr` required? Won't `&modelValues` work?

Reply:
right, it is not constant, so `&modelValues` just work